### PR TITLE
Reintroduce 'SleepImpl' sleep implementation

### LINKED ISSUE #1962: Recent `sf::Time` changes result in choppy gameplay with frametime limit enabled
PR #1872 resulted in dramatic performance degradation (multiple reports of choppiness, unstable FPS, input lag) in Open Hexagon when `Window::setFramerateLimit` is used due to commit db83f857feeacaa30a482aa6059b9358e9ccb2ef. Reverting that commit fixes the issue. 

I am not entirely sure what the problem is, but `sf::sleep` seems to be very inaccurate with that commit, possibly related to #1946. For example, setting an FPS limit to 60 results in very varied FPS between 50-64, while with that commit reverted it is quite accurate. I think we might have to revert this one once we figure out exactly why it is inaccurate and if that can be fixed.

### PR DESCRIPTION
**Pull request by vittorioromeo**
Fixes #1962.

### CODE REVIEW

**Comment by Bromeon** [comment]
Is `this_thread::sleep_for()` only broken on MinGW, or Windows in general?

One option would be to use standard implementation on those platforms where it's reliable, but I also see that this may be an unnecessary risk for bugs/regressions in the future. Maybe we should just stick to something we know to work.

**Comment by vittorioromeo** [comment]
> Is `this_thread::sleep_for()` only broken on MinGW, or Windows in general?

This problem is specific to the `winpthreads` library that is bundled with any MinGW-w64 distribution, including MSYS2, Nuwen, and the CodeBlocks ones, which are the most popular. I don't think it's "broken", it's just not implemented optimally. For example,  
a [one-line change](https://stackoverflow.com/a/54495870/598696) might fix the issue, but I have not tried it yet. It also seems that [Visual Studio/MSVC suffers from similar issues](https://stackoverflow.com/a/32963363/598696).

> One option would be to use standard implementation on those platforms where it's reliable, but I also see that this may be an unnecessary risk for bugs/regressions in the future. Maybe we should just stick to something we know to work.

I don't really know how to test reliability. I think that the latter is a better option, unless someone else wants to try other possibilities.

**Comment by vittorioromeo** [comment]
I did some more research and I'm not sure this PR is a good idea. Check https://github.com/SFML/SFML/issues/1962.

**Comment by vittorioromeo** [comment]
I did some more testing/research and this PR is probably the best way to go if we don't want to change how `setFramerateLimit` works.

File: src/SFML/System/Win32/SleepImpl.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+////////////////////////////////////////////////////////////
+void sleepImpl(Time time)
+{
+    // Get the minimum supported timer resolution on this system
+    static const UINT periodMin = []
+    {
+        TIMECAPS tc;
+        timeGetDevCaps(&tc, sizeof(TIMECAPS));
+        return tc.wPeriodMin;
+    }();
```
Comment (eXpl0it3r) [inline src/SFML/System/Win32/SleepImpl.cpp:43]: I assume this is to reduce the calls to `timeGetDevCaps` right?

**Review by eXpl0it3r — APPROVED** [review]

File: src/SFML/System/Win32/SleepImpl.cpp (hunk 1 above)
Comment (vittorioromeo) [reply src/SFML/System/Win32/SleepImpl.cpp:43]: Yes, this was originally suggested by @reconndev. It's not really about performance, but more about communicating the fact that `wPeriodMin` will never change -- it's hardware-specific.

**Pull request merged**